File: base/android/java/src/org/chromium/base/metrics/StatisticsRecorderAndroid.java
// Copyright 2016 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.base.metrics;

import org.jni_zero.JNINamespace;
import org.jni_zero.JniType;
import org.jni_zero.NativeMethods;

/** Java API which exposes the registered histograms on the native side as JSON test. */
@JNINamespace("base::android")
public final class StatisticsRecorderAndroid {
    private StatisticsRecorderAndroid() {}

    /**
     * @param verbosityLevel controls the information that should be included when dumping each of
     * the histogram.
     * @return All the registered histograms as JSON text.
     */
    public static String toJson(@JSONVerbosityLevel int verbosityLevel) {
        return StatisticsRecorderAndroidJni.get().toJson(verbosityLevel);
    }

    @NativeMethods
    interface Natives {
        @JniType("std::string")
        String toJson(@JSONVerbosityLevel @JniType("JSONVerbosityLevel") int verbosityLevel);
    }
}

File: base/android/statistics_recorder_android.cc
// Copyright 2016 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/metrics/statistics_recorder.h"

#include <string>

#include "base/android/jni_string.h"
#include "base/metrics/histogram_base.h"
#include "base/system/sys_info.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "base/metrics_jni/StatisticsRecorderAndroid_jni.h"

namespace base {
namespace android {

static std::string JNI_StatisticsRecorderAndroid_ToJson(
    JNIEnv* env,
    JSONVerbosityLevel verbosityLevel) {
  return base::StatisticsRecorder::ToJSON(verbosityLevel);
}

}  // namespace android
}  // namespace base
